services/platform/platform.go

```diff
@@ -192,6 +194,13 @@ func (s *Service) refreshSession(ctx context.Context) error {
 
 	_, err = api.NewAuthAPIClient(cc).RefreshSession(ctx, &api.RefreshSessionRequest{})
 	if err != nil {
+		// If SaaS credentials become invalid then force a logout so that the next
+		// refresh session attempt is successful.
+		logoutErr := saasdial.LogoutIfInvalidAuth(s.db, s.l, err)
+		if logoutErr != nil {
+			return errors.Wrap(fmt.Errorf("%v %v", err, logoutErr), "failed to refresh session")
```

Comment:
It will be a really strange error message I suppose. Maybe better just to log `logoutErr` here instead of attaching it to the `err`?